<p>Using trailing slashes for collections and not for documents makes it more difficult for users to know how
to build the correct URL. Consistently use paths without a trailing slash to give your API a more consistent look
and increase its accessibility.</p>
<h2>Noncompliant Code Example</h2>
<pre>
/v1/transaction-codes/
</pre>
<h2>Compliant Solution</h2>
<pre>
/v1/transaction-codes
</pre>

