<p>Spinal case favors readability of the URL by humans, by clearly separating meaningful words. Using Camel case is
discouraged, since it forces the end user to remember where to put the capital letters. This is true in particular
for acronyms, where there is no consistent rule on wether or not to use capital case for all letters of the acronym.</p>
<p>Using only lower-case characters and dashes to separate words reduces brain overload for the end user and increases
the consistency, hence the accessibility, of your API.</p>
<h2>Noncompliant Code Example</h2>
<pre>
/v1/transactionCodes
</pre>
<h2>Compliant Solution</h2>
<pre>
/v1/transaction-codes
</pre>

